torch_np/_wrapper.py

```diff
@@ -369,9 +446,25 @@ def concatenate(ar_tuple, axis=0, out=None, dtype=None, casting="same_kind"):
     return _helpers.result_or_out(result, out)
 
 
-@asarray_replacer()
 def bincount(x, /, weights=None, minlength=0):
-    return torch.bincount(x, weights, minlength)
+    if not isinstance(x, ndarray) and x == []:
+        # edge case allowed by numpy
+        x = asarray([], dtype=int)
+
+    x_tensor = asarray(x).get()
```

Comment:
What about using the nice and newly added `to_tensors_or_none` on `x` and `weights`.